frontend/app/src/components/status/AppUpdateIndicator.vue

```diff
@@ -10,61 +10,60 @@
 </template>
 
 <script lang="ts">
-import { Component, Vue, Watch } from 'vue-property-decorator';
-import { mapActions, mapGetters, mapState } from 'vuex';
-import { Version } from '@/store/types';
+import { computed, defineComponent, toRefs, watch } from '@vue/composition-api';
+import { useInterop } from '@/electron-interop';
+import { useMainStore } from '@/store/store';
+import { useStore } from '@/store/utils';
 
-@Component({
-  computed: {
-    ...mapState(['version']),
-    ...mapGetters('settings', ['versionUpdateCheckFrequency']),
-    ...mapGetters(['updateNeeded'])
-  },
-  methods: {
-    ...mapActions('session', ['openUpdatePopup'])
-  }
-})
-export default class AppUpdateIndicator extends Vue {
-  updateNeeded!: boolean;
-  version!: Version;
-  versionUpdateCheckFrequency!: number;
-  refreshInterval: any;
+export default defineComponent({
+  name: 'AppUpdateIndicator',
+  setup() {
+    let refreshInterval: any = undefined;
+    const mainStore = useMainStore();
+    const store = useStore();
+    const { version, updateNeeded } = toRefs(mainStore);
```

Comment:
Just want to ask, is `toRefs` and `storeToRefs` doing the same thing. If yes, can we make an agreement to only use one of them?

Reply:
Ideally, we would use `storeToRefs` but there is a bug when used with vue 2 where it doesn't work for computed properties, it only works with `ref()`. 

Since in the future we will use `storeToRefs` either after upgrading to vue 3 or if the bug gets fixed as a rule of thumb if you just want to use refs from the store use `storeToRefs` or fall back to `toRefs` when you need to access computed properties.

Reply:
I see, noted